File: client/src/domain.ts
export interface BooksById {
  [uuid: string]: Book;
}

export interface Book {
  gutenbergId: number;
  id: Uuid;
  author: Author;
  title: BookTitle;
  genres: Genre[];
}

export type Uuid = string;

export interface Author {
  firstName: string;
  lastName: string;
  birthYear: number;
  deathYear: number | null;
}

export interface BookTitle extends TranslatedContent {}

export interface Genre {
  name: TranslatedContent;
}

export interface TranslatedContent {
  [lang: string]: string;
}

File: src/library_import/domain.py
import typing as t
from enum import Enum


class BookAssetType(Enum):
    EPUB = "epub"
    MOBI = "mobi"


class BookAsset(t.NamedTuple):
    type: BookAssetType
    size: int


class Book(t.NamedTuple):
    gutenberg_id: int
    title: str
    lang: str
    genres: t.List[str]
    assets: t.List[BookAsset]


class Author(t.NamedTuple):
    gutenberg_id: int
    name: str
    first_name: str = None
    last_name: str = None
    birth_year: int = None
    death_year: int = None
